modules/signals/rxjs-interop/src/rx-method.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  assertInInjectionContext,
+  DestroyRef,
+  effect,
+  inject,
+  Injector,
+  isSignal,
+  Signal,
+  untracked,
+} from '@angular/core';
+import { isObservable, Observable, of, Subject, Unsubscribable } from 'rxjs';
+
+type RxMethodInput<Input> = Input | Observable<Input> | Signal<Input>;
+
+type RxMethod<Input> = ((input: RxMethodInput<Input>) => Unsubscribable) &
+  Unsubscribable;
+
+export function rxMethod<Input>(
+  generator: (source$: Observable<Input>) => Observable<unknown>,
+  config?: { injector?: Injector }
+): RxMethod<Input> {
+  if (!config?.injector) {
+    assertInInjectionContext(rxMethod);
+  }
+
+  const injector = config?.injector ?? inject(Injector);
+  const destroyRef = injector.get(DestroyRef);
+  const source$ = new Subject<Input>();
+
+  const sourceSub = generator(source$).subscribe();
+  destroyRef.onDestroy(() => sourceSub.unsubscribe());
+
+  const rxMethodFn = (input: RxMethodInput<Input>) => {
+    let input$: Observable<Input>;
+
+    if (isSignal(input)) {
+      input$ = toObservable(input, injector);
+    } else if (isObservable(input)) {
+      input$ = input;
+    } else {
+      input$ = of(input);
+    }
+
+    const instanceSub = input$.subscribe((value) => source$.next(value));
+    sourceSub.add(instanceSub);
+
+    return instanceSub;
+  };
+  rxMethodFn.unsubscribe = sourceSub.unsubscribe.bind(sourceSub);
+
+  return rxMethodFn;
+}
+
+/**
+ * A custom `toObservable` function that uses `Subject`
+ * instead of `ReplaySubject` to reduce the bundle size.
+ */
+function toObservable<T>(source: Signal<T>, injector: Injector): Observable<T> {
```

Comment:
Wouldn't this end up with an increase when applications already make use of the official `toObservable` ?
And/or, should we then expose this method?

Reply:
Yes, you're right. It left in the playground project when I did some experiments related to the bundle size.

I'll remove a custom `toObservable` function.